crates/dojo/types/src/schema.rs

```diff
@@ -308,6 +312,196 @@ impl Ty {
             }
         }
     }
+
+    /// Convert a Ty to a JSON Value
+    pub fn to_json_value(&self) -> Result<JsonValue, PrimitiveError> {
+        match self {
+            Ty::Primitive(primitive) => match primitive {
+                Primitive::Bool(Some(v)) => Ok(json!(*v)),
+                Primitive::I8(Some(v)) => Ok(json!(*v)),
+                Primitive::I16(Some(v)) => Ok(json!(*v)),
+                Primitive::I32(Some(v)) => Ok(json!(*v)),
+                Primitive::I64(Some(v)) => Ok(json!(v.to_string())),
+                Primitive::I128(Some(v)) => Ok(json!(v.to_string())),
+                Primitive::U8(Some(v)) => Ok(json!(*v)),
+                Primitive::U16(Some(v)) => Ok(json!(*v)),
+                Primitive::U32(Some(v)) => Ok(json!(*v)),
+                Primitive::U64(Some(v)) => Ok(json!(v.to_string())),
+                Primitive::U128(Some(v)) => Ok(json!(v.to_string())),
+                Primitive::USize(Some(v)) => Ok(json!(*v)),
+                Primitive::U256(Some(v)) => {
+                    let bytes = v.to_be_bytes();
+                    let high = u128::from_be_bytes(bytes[..16].try_into().unwrap());
+                    let low = u128::from_be_bytes(bytes[16..].try_into().unwrap());
+                    Ok(json!({
+                        "high": high.to_string(),
+                        "low": low.to_string()
+                    }))
+                }
+                Primitive::Felt252(Some(v)) => Ok(json!(format!("{:#x}", v))),
+                Primitive::ClassHash(Some(v)) => Ok(json!(format!("{:#x}", v))),
+                Primitive::ContractAddress(Some(v)) => Ok(json!(format!("{:#x}", v))),
```

Comment:
We should in the future have this format being factorized. I always asking myself if it needs to be padded or not. x) Having this in functions will help for clearer context across torii/the stack.